src/scripts/main.js

```diff
@@ -1 +1,84 @@
 'use strict';
+
+const button = document.querySelector('body');
+
+button.addEventListener('contextmenu', (e) => {
+  e.preventDefault();
+});
+
+const firstPromise = new Promise((resolve, reject) => {
+  setTimeout(reject, 3000);
+
+  button.addEventListener('click', resolve);
+});
+
+const secondPromise = new Promise((resolve) => {
+  button.addEventListener('mouseup', (e) => {
+    switch (e.button) {
+      case 0:
+        resolve();
+        break;
+      case 2:
+        e.preventDefault();
+        resolve();
+        break;
+    }
+  });
+});
+
+const thirdPromise = new Promise((resolve) => {
+  let leftClick = false;
+  let rightClick = false;
+
+  const checkClicks = () => {
+    if (leftClick && rightClick) {
+      resolve();
+    }
+  };
+
+  button.addEventListener('mouseup', (e) => {
+    if (e.button === 0) {
+      leftClick = true;
+    } else if (e.button === 2) {
+      e.preventDefault();
+      rightClick = true;
```

Comment:
Similar to the previous comment, `e.preventDefault()` is used here to prevent the default context menu on right-click. Confirm if this behavior is desired.